### ISSUE TITLE
train-model workflow failed

### ISSUE THREAD

**Issue by dthaler** [orcasound/pods-ai#223]
https://github.com/orcasound/pods-ai/actions/runs/25969248292/job/76338152537 failed.

The failure is caused by loading a pretrained classifier head that does not match this training job’s label count.

### Root cause
In `src/train_podsai_model.py` the model is loaded here:

- `src/train_podsai_model.py:466`
- `src/train_podsai_model.py:468-470`

```python
model = AutoModelForAudioClassification.from_pretrained(
    args.model_name,
    num_labels=len(LABEL2ID),
    label2id=LABEL2ID,
    id2label=ID2LABEL,
)
```

The workflow runs training with the default `--num_classes 7`, so `num_labels=len(LABEL2ID)` becomes 7.  
But the base model `MIT/ast-finetuned-audioset-10-10-0.4593` has a pretrained classification head for a different number of labels. Transformers rejects that mismatch because `ignore_mismatched_sizes=False` by default.

The log points directly to this failure in:
- `src/train_podsai_model.py:466-476`
- workflow: `.github/workflows/train_model.yml`

### Fix
Tell Transformers to discard and reinitialize the incompatible classification head while keeping the pretrained backbone.

Update the model load call to:

```python
model = AutoModelForAudioClassification.from_pretrained(
    args.model_name,
    num_labels=len(LABEL2ID),
    label2id=LABEL2ID,
    id2label=ID2LABEL,
    ignore_mismatched_sizes=True,
)
```

### Suggested patch
In `src/train_podsai_model.py` around line 466:

```python
try:
    model = AutoModelForAudioClassification.from_pretrained(
        args.model_name,
        num_labels=len(LABEL2ID),
        label2id=LABEL2ID,
        id2label=ID2LABEL,
        ignore_mismatched_sizes=True,
    )
except Exception as e:
    error_msg = f"Error loading model from {args.model_name}: {type(e).__name__}: {e}"
    print(error_msg)
    print("Please ensure the model name is correct and you have internet connectivity.")
    raise RuntimeError(error_msg) from e
```

### Why this is the right fix
This training script is explicitly fine-tuning a base model for a new task with either:
- 2 labels, or
- 7 labels

That means the classifier layer often must be replaced. `ignore_mismatched_sizes=True` is the standard way to do that for Hugging Face classification models.

### Optional improvement
To make the intent clearer in logs, add a comment:

```python
# Reinitialize the classification head if the pretrained model's label count
# differs from this task's label set.
```

### Relevant files
- Failing code: [`src/train_podsai_model.py` at `d3893e6b66669f66763f2ef3dbf81081dcaa68fc`](https://github.com/orcasound/pods-ai/blob/d3893e6b66669f66763f2ef3dbf81081dcaa68fc/src/train_podsai_model.py#L466-L476)
- Job definition: [`.github/workflows/train_model.yml` at `d3893e6b66669f66763f2ef3dbf81081dcaa68fc`](https://github.com/orcasound/pods-ai/blob/d3893e6b66669f66763f2ef3dbf81081dcaa68fc/.github/workflows/train_model.yml#L71-L78)

If you want the minimal change, this is a one-line fix: add `ignore_mismatched_sizes=True` to the `from_pretrained(...)` call.

Also, can a test be added that would have caught this before?
